pkg/upgraders/controlplanestep.go

```diff
@@ -43,14 +43,22 @@ func (c *clusterUpgrader) ControlPlaneUpgraded(ctx context.Context, logger logr.
 
 	isCompleted := c.cvClient.HasUpgradeCompleted(clusterVersion, c.upgradeConfig)
 	history := cv.GetHistory(clusterVersion, c.upgradeConfig.Spec.Desired.Version)
+	var upgradeStartTime time.Time
+	var controlPlaneCompleteTime time.Time
 	if history == nil {
-		return false, err
+		upgradeStartTime, err = time.Parse(time.RFC3339, c.upgradeConfig.Spec.UpgradeAt)
+		if err != nil {
+			return false, err //error parsing time string
+		}
+	} else {
+		upgradeStartTime = history.StartedTime.Time
```

Comment:
I think you may need to do a check to verify that `history.StartedTime` is not nil / `IsZero` before you go accessing the `Time` in it or else this might cause a panic.

Reply:
Note that if the StartedTime is nil, we would still want the code logic to be able to set the timeout metric.
ie. if there is never a startedtime and the upgrade window elapses, we still need to set that timeout metric.